solr/core/src/java/org/apache/solr/search/SolrQueryTimeoutImpl.java

```diff
@@ -45,7 +45,6 @@ public static Long getTimeoutAtNs() {
     return timeoutAt.get();
   }
 
-  @Override
```

Comment:
This innocuous looking removal was very curious to me and I went digging into Lucene's changes that led to this.  Based on what I learned, I predict a performance degradation for any query that isn't using timeouts.  The Lucene issue is https://github.com/apache/lucene/issues/11914 and it means that if Solr does nothing about this, then ExitableDirectoryReader.wrap (used in SolrIndexSearcher) will wrap all underlying low level index components (Terms, ...) because Lucene's ExitableDirectoryReader no longer guards against this with a "isTimeoutEnabled" check as it no longer exists.  Solr could fork the old ExitableDirectoryReader, which would be sad but pretty easy.  Alternatively, maybe SolrIndexSearcher could only wrap for the execution of a query -- not sure as I haven't explored this.

Reply:
Thanks, @dsmiley helping us here.
didn't understand fully, but will try again :)

Reply:
hi @dsmiley, can we address this issue in separate Jira or should we include it in this only?

Reply:
Okay, let's.  It may be a release blocker; hopefully there are benchmarks that can show yes/no.  FYI @chatman

Reply:
Hi @dsmiley, any update on this one?